mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/binding/impl/ValueBindingContextImpl.java

```diff
@@ -188,7 +189,12 @@ private <F> IndexFieldTypeOptionsStep<?, F> inferFieldType(ValueBridge<?, F> bri
 				.orElseThrow( () -> new AssertionFailure( "Could not auto-detect the return type for value bridge '"
 						+ bridge + "'." ) );
 		if ( typeArgument instanceof Class ) {
-			return indexFieldTypeFactory.as( (Class<F>) typeArgument );
+			if ( bridge instanceof VectorBridge ) {
+				return indexFieldTypeFactory.asVector( (Class<F>) typeArgument );
+			}
```

Comment:
If bridges want to use a specific field type, they should use a [value binder](https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#binding-valuebridge-valuebinder), so this shouldn't be necessary.

Why didn't you end up using the `FieldModelContributor` to change the default behavior depending on whether `@VectorField` or another annotation isused, as I suggested?

Reply:
> Why didn't you end up using the FieldModelContributor to change the default behavior depending on whether @VectorField or another annotation isused, as I suggested?

I'm not sure I got your idea correctly 😃, but I've made some changes in this commit here https://github.com/hibernate/hibernate-search/pull/3814/commits/4336a365d28285797e8c4b8be9f9874c614c87bb let me know if that's what you had in mind 😃

Reply:
Close enough, here's what I had in mind: https://github.com/marko-bekhta/hibernate-search/pull/215
Either will work though :shrug: